pvlib/pvsystem.py

```diff
@@ -223,12 +251,49 @@ def __init__(self,
             )
 
     def __repr__(self):
-        attrs = ['name', 'surface_tilt', 'surface_azimuth', 'module',
-                 'inverter', 'albedo', 'racking_model', 'module_type',
-                 'temperature_model_parameters']
-        return ('PVSystem:\n  ' + '\n  '.join(
-            f'{attr}: {getattr(self, attr)}' for attr in attrs))
+        repr = f'PVSystem:\n  name: {self.name}\n  '
+        for array in self._arrays:
+            repr += '\n  '.join(array.__repr__().split('\n'))
+            repr += '\n  '
+        repr += f'inverter: {self.inverter}'
+        return repr
+
+    def _validate_per_array(self, values, system_wide=False):
+        """Check that `values` is a tuple of the same length as
+        `self._arrays`.
+
+        If it is a single vlaue it is packed in to a length-1 tuple before
+        the check. If the length is not the same a ValueError is raised,
+        otherwise the tuple is returned.
```

Comment:
```suggestion
        If `values` is not a tuple it is packed in to a length-1 tuple before
        the check. If the lengths are not the same a ValueError is raised,
        otherwise the tuple `values` is returned.
```